internal/daemon/worker/controller_connection.go

```diff
@@ -78,6 +78,7 @@ func (w *Worker) controllerDialerFunc() func(context.Context, string) (net.Conn,
 				nodeauth.MakeCurrentParametersFactory(
 					ctx,
 					nodee.NopTransactionStorage(w.NodeeFileStorage),
+					nodee.WithWrapper(w.conf.WorkerStorageKms),
```

Comment:
what happens if the storage wrapper is nil (is it an error we can detect before returning)?  Also, is it possible to write a unit test for this?

Reply:
Do we want to ensure this is not nil? In the [nodee library](https://github.com/hashicorp/nodeenrollment/blob/2f27f4add8d11675c784a586e26347dd6f2a71af/types/node_credentials.go#L62) it just stores without encryption if this wrapper is nil

Reply:
I guess do we want to throw an error any time a worker storage KMS wrapper is not passed in?

Reply:
while I understand why the nodee library wants to support not encrypting creds, does boundary want to support that workflow?  sort of a broader question you might consider asking the team.

Reply:
I think we need to, yes. Not everyone will have access to a suitable KMS, or may have other possibilities (an encrypted USB drive that's inserted and unlocked as long as it has power; a ramdisk; ephemeral KMS workflow; dev/test environments; etc.